### PR TITLE
Natspec IFeeCurrencyAdapter

### PR DESCRIPTION
**Pull request by martinvol**
### Description

_A few sentences describing the overall effects and goals of the pull request's commits.
What is the current behavior, and what is the updated/expected behavior with this PR?_

### Other changes

_Describe any minor or "drive-by" changes here._

### Tested

_An explanation of how the changes were tested or an explanation as to why they don't need to be._

### Related issues

- Fixes #[issue number here]

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

### Documentation

_The set of community facing docs that have been added/modified because of this change_

### CODE REVIEW

File: packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol [hunk 1]
```diff
@@ -2,22 +2,51 @@
...
+   */
   function getAdaptedToken() external view returns (address);
 
+  /**
+   * @return The multiplier that should be used when upscaling and downscaling.
+   */
   function digitDifference() external view returns (uint96);
 
+  /**
+   * @return returns the amount that it's debited after calling debitGasFees().
```
Comment (marekolszewski) [inline packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol]: I would change to: returns the amount that *is* debited after calling debitGasFees() and before creditGasFees().

File: packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol [hunk 2]
```diff
@@ -2,22 +2,51 @@
...
+
+  /**
+   * @return same as decimals.
+   */
   function expectedDecimals() external view returns (uint8);
 
-  function decimals() external view returns (uint8);
 
+  /**
+   * @notice See IFeeCurrency.
```
Comment (marekolszewski) [inline packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol]: I would explicitly say that these are like the ones in IFeeCurrency, but using the number of decimals native to this adapter.

File: packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol [hunk 3]
```diff
@@ -2,22 +2,51 @@
...
+   */
   function debitGasFees(address from, uint256 value) external;
 
+
+  /**
+   * @notice See IFeeCurrency.
+   */
```
Comment (marekolszewski) [inline packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol:49]: Same as above.

**Review by marekolszewski — CHANGES_REQUESTED** [review]
Left some comments.

File: packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol [hunk 4]
```diff
@@ -2,22 +2,51 @@
 pragma solidity >=0.8.7 <0.8.20;
 
 interface IFeeCurrencyAdapter {
+  /**
+   * @return The address of the adapted token.
+   */
   function getAdaptedToken() external view returns (address);
 
+  /**
+   * @return The multiplier that should be used when upscaling and downscaling.
```
Comment (marekolszewski) [inline packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol]: Can we explicitly say that this is: 10**(wrapperDigits - tokenDigits)?

Comment (martinvol) [inline packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol]:
```suggestion
   * @return The multiplier that should be used when upscaling and downscaling. This is the result of 10**(expectedDecimals - getAdaptedToken().decimals()).
```

File: packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol [hunk 5]
```diff
@@ -2,22 +2,51 @@
...
+   */
   function getAdaptedToken() external view returns (address);
 
+  /**
+   * @return The multiplier that should be used when upscaling and downscaling. This is the result of 10**(expectedDecimals - getAdaptedToken().decimals()).
+   */
   function digitDifference() external view returns (uint96);
 
+  /**
+   * @return returns the amount that it's debited after calling debitGasFees().
```
Comment (martinvol) [inline packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol]:
```suggestion
   * @return The amount that is debited after calling debitGasFees() and before creditGasFees().
```

File: packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol [hunk 6]
```diff
@@ -2,22 +2,51 @@
...
+
+  /**
+   * @return same as decimals.
+   */
   function expectedDecimals() external view returns (uint8);
 
-  function decimals() external view returns (uint8);
 
+  /**
+   * @notice See IFeeCurrency.
```
Comment (martinvol) [inline packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol]:
```suggestion
   * @notice As debitGasFees in IFeeCurrency, always using the number of decimals the evm expects.
```

File: packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol [hunk 7]
```diff
@@ -2,22 +2,51 @@
...
-  function decimals() external view returns (uint8);
 
+  /**
+   * @notice As debitGasFees in IFeeCurrency, always using the number of decimals the evm expects.
+   */
   function debitGasFees(address from, uint256 value) external;
 
+
+  /**
+   * @notice See IFeeCurrency.
```
Comment (martinvol) [inline packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol]:
```suggestion
   * @notice Same as creditGasFees in IFeeCurrency, always using the number of decimals the evm expects.
```

File: packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol [hunk 8]
```diff
@@ -2,22 +2,51 @@
...
+
+  /**
+   * @return same as decimals.
+   */
   function expectedDecimals() external view returns (uint8);
 
-  function decimals() external view returns (uint8);
 
+  /**
+   * @notice As debitGasFees in IFeeCurrency, always using the number of decimals the evm expects.
```
Comment (martinvol) [inline packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol]:
```suggestion
   * @notice Same as debitGasFees in IFeeCurrency, always using the number of decimals the evm expects.
```

**Review by soloseng — APPROVED** [review]

**Pull request closed without merging**